packages/dialect-react-sdk/context/DialectContext/Sdk/index.ts

```diff
@@ -22,7 +24,33 @@ function useDialectSdk(
     encryptionKeysStore,
     backends = EMPTY_ARR,
   } = config;
-  const { adapter, connected } = DialectWallet.useContainer();
+  const { adapter, connected, initiateConnection } =
+    DialectWallet.useContainer();
+
+  // checks if wallet is already authorized to skip not authorized screen
+  useEffect(
+    function preValidateSdkToken() {
+      if (!dialectCloud) return;
+      if (!dialectCloud.tokenStore) return;
+      if (!adapter.publicKey) return;
+      // extract token store
+      let tokenStore: TokenStore;
+      if (typeof dialectCloud.tokenStore === 'string') {
+        if (dialectCloud.tokenStore !== 'local-storage') return;
+        tokenStore = TokenStore.createLocalStorage();
```

Comment:
I'm not fully understanding what' going on here. Could you elaborate either here or by adding comments, please?

Reply:
good point, added description, logic is a bit tricky